pkg/controller/integration/error.go

```diff
@@ -56,6 +57,27 @@ func (action *errorAction) Handle(ctx context.Context, integration *v1.Integrati
 		return integration, nil
 	}
 
+	// the integration in error state may have recovered and the running pod may be ok
+	// at this point we need to check, if the pod is ok set the integration to running.
+	podList, _ := action.client.CoreV1().Pods(integration.Namespace).List(ctx, metav1.ListOptions{
```

Comment:
The reconcile loop won't always be called, as Pods are not being watched by the Integration reconciler.